client/app/pages/queries/QuerySource.jsx

```diff
@@ -213,7 +213,7 @@ function QuerySource(props) {
                 <DynamicComponent
                   name={"QuerySourceDropdown"}
                   dataSources={dataSources}
-                  value={dataSource ? dataSource.id : undefined}
+                  value={dataSource ? String(dataSource.id) : undefined}
```

Comment:
This seems to cause bugs to percy:

<img width="1771" alt="Screen Shot 2020-11-24 at 09 43 39" src="https://user-images.githubusercontent.com/3356951/100095615-8ab40800-2e39-11eb-9d2a-2d8955959c5d.png">

Reply:
How about just update https://github.com/getredash/redash/blob/9097feb100b10e022df16e3c7dc3f490754a2f95/client%2Fapp%2Fpages%2Fqueries%2Fcomponents%2FQuerySourceDropdown.jsx#L32

to be `PropTypes.oneOfType([PropTypes.string, PropTypes.number])`? Considering it is supposed to be the id and we need to support both types, that should do it 🙂